console-err-msg

Rispectech opened the pull request:
In order to prevent redundancy in each component, I created a new js file (errorfunc.js) which contains all the error functions I used apart from occasional exceptions where it couldn't be used 

in errorfunc.js , checkMissingValueObj checks whether the values are present or not

kartikcho commented on src/components/educationCard/EducationCard.js:
```diff
@@ -16,6 +20,9 @@ export default function EducationCard({school}) {
   const {isDark} = useContext(StyleContext);
+
+  if (!school.logo)
+    console.log(`Image of schoolName ${school.name} is not available`);
```
We can keep this in as it seems helpful but change it to `console.error('Image of ${school.name} is missing in education section');`

kartikcho commented on src/components/errorfunc.js:
```diff
@@ -0,0 +1,48 @@
+//function to check whether the url is correct
```
You'll probably need to scrap all of this code and it's usage in other components as it isn't what is required currently :(

kartikcho reviewed (changes requested):
I took a quick look over the changes and it seems you understood the issues incorrectly. The issue #212 says to add error messages for only third party components, i.e GitHub repos, Twitter embed, podcasts and the GitHub contact components.

Throwing errors for missing fields would be counter intuitive as we also want users to have the option to keep fields blank if they wish to do so. Lots of errors in the console blowing up in people's face will make them feel they're doing something wrong when it's just an empty field.

Also looking at the implementation, you're using logs instead of errors which isn't the best way to warn someone.

Rispectech commented:
> I took a quick look over the changes and it seems you understood the issues incorrectly. The issue #212 says to add error messages for only third party components, i.e GitHub repos, Twitter embed, podcasts and the GitHub contact components.
> 
> Throwing errors for missing fields would be counter intuitive as we also want users to have the option to keep fields blank if they wish to do so. Lots of errors in the console blowing up in people's face will make them feel they're doing something wrong when it's just an empty field.
> 
> Also looking at the implementation, you're using logs instead of errors which isn't the best way to warn someone.

okay, so you want the error using throw error instead of log.right will do this one 
second right using the error function for all the components intuitive so fix this as well 
and how about editing the error function to throw error instead of console message for missing  values in repos card
and a doubt can u please tell how the rest of components fetch data ( Twitter embeds , podcast )

Rispectech commented:
i have converted the console messages in third party components to error messages and removed them from the rest 
errorfunc.js deleted as well
please check and review the changes
thanks

Rispectech commented:
sorry @kartikcho for being such a pain making you request changes again and again 
I got confused with throw and console when you last commented so made these mistakes so sorry for that 
I have made the changes so please review and check 
thanks

kartikcho commented:
@allcontributors add @Rispectech for code
